### ISSUE TITLE
Tag editing when downloading audio

### ISSUE THREAD

**Issue by fugounashi** [TeamNewPipe/NewPipe#183]
I guess YouTube doesn't provide structured artist/album/track fields so it'd be nice if there was a simple tag editor to set that info when downloading.

**Comment by theScrabi** [comment]
I'm not going to implement something like this. But if you find a java lib that could do so, we could speak about this feature.

**Comment by monnier** [comment]
FWIW, this is my main "wishlist item" for NewTube, indeed.

**Comment by wb9688** [comment]
@theScrabi You can use jaudiotagger...

@monnier What is NewTube?

**Comment by theScrabi** [comment]
@wb9688 all right thanks. The only problem so far is, that I'd neet to implement it. But you might understand, that there are million other things which come first. So it also won't happen the next time, unless some of you guys help me :)

**Comment by fugounashi** [comment]
@theScrabi 

It seems the /Music structure is only /Music/album/track, so to get artist as well tags are needed.

As for libraries, I came across this: http://stackoverflow.com/questions/9707572/how-to-get-and-set-change-id3-tag-metadata-of-audio-files

there are a few tagging apps in the play store as a workaround

thanks!

**Comment by theScrabi** [comment]
I think that would be something that should come first. A taging app on f-droid :P

**Comment by hydroid7** [comment]
It is a have to to use _ characters for filenames? Or can we replace _ with " "?

Or it would be an opinion to ask for a filename before downloading it?

**Comment by Wiredframe** [comment]
I'd also like to ask you to add at least SUPPORT for the M4a files to be tagged by another app. Right now it's not working because "This M4a file doesn't support ID3 tagging at all".

I hope it's possible because right now it's awful to convert m4a to mp3 just to tag them. ^^ 

As always, THANKS in advance!

**Comment by aFoxWhoIsNice** [comment]
Please add this feature. It would be fantastic.

**Comment by wb9688** [comment]
@Wiredframe: Every container has its own tagging format. ID3 is MP3's
tagging format, which ofcourse can't be used with M4A. You need to use an
app that supports tagging M4A.

**Comment by Wiredframe** [comment]
The problem is that many apps say that it's not a valid m4a file, I guess the header is missing or something. But I found an app that can tag even those files, it's called AutomaTag.

**Comment by wb9688** [comment]
@Wiredframe: I don't know, but we're just downloading it straight from YouTube.

**Comment by Wiredframe** [comment]
Thanks again. I guess this issue should be closed since there's nothing to do here ^^

**Comment by theScrabi** [comment]
Oh well keep it open, you never know.

**Comment by albjeremias** [comment]
there could be a simplex regex to get most of the tags automatically... and show like advanced so user could edit the regex? or edit the automatic tag... also https://en.m.wikipedia.org/wiki/MusicBrainz could be used later...

**Comment by SkyzohKey** [comment]
What about using the spotify API to get tags ? We can add this to the scrapper so that it is able to return tags for a YT video and apply them once download is finished I guess.

**Comment by SkyzohKey** [comment]
Don't close this, bot. 😉

**Comment by Stypox** [comment]
Another way to get song name and author is [this](https://github.com/Stypox/song-downloader/blob/master/song-downloader/song-downloader.py#L59).
From the readme:
> Imagine a playlist (id ``1234567890abcdefghijklmnopqrstuvwx``) whose second video (id ``0a1b2c3d4e5``) is titled ``ArtistName feat. Ft1 | SongName [Rel release](Remixer ft. Ft2 remix)``. [...] the song's ID3 metadata would be:

    artist: "Remixer (original by ArtistName)"
    title: "SongName (remix)"
    tracknumber: "1"
    album: "1234567890abcdefghijklmnopqrstuvwx"

> "tracknumber" is **1** because the video is the **second** one of the playlist

**Comment by Wilker-a** [comment]
i think it would be very useful for SoundCloud, where the album image and year of publishing should also be taken into consideration.

in addition to that (but a topic for another issue) is that i think that SoundCloud media should have something showing if the song is originally a free to download, a paid one, or if there was no link at all in the site, and the application should give the option to block or redirect(with a note in the download button text) downloads of songs that are paid.

**Comment by MakoriSM** [comment]
The ability to write your own artist and song name is definitely my most desired feature. 
It would also be great if we could use the youtube thumbnail as album art

Newpipe looks like the best alternative to the now defunct Peggo but I hate having my music library filled with songs titled like: "verylongsongname (lyrics) - artist" with the artist set to: "unknown artist"

**Comment by opusforlife2** [comment]
> I guess YouTube doesn't provide structured artist/album/track fields so it'd be nice if there was a simple tag editor to set that info when downloading.

It does (youtube-dl is capable of parsing it and adding it to the download), so I'm closing this in favour of #4341.
